### PR TITLE
OVEP: Tensor caching fix

### PR DESCRIPTION
**Pull request by saurabhkale17**
### Description
1. changing the emplace to [] that does have a difference, emplace will only create a new entry if it doesn't already exist in the map
2. change the logic of the caching lookup to key off of input/output names instead of ort raw ptrs.
3. changes OV tensor creation for CPU allocated input/output ORT tensors. The CPU allocated input/output tensor path was re-allocating OV tensors based on the ORT input/output tensors. So we'd get 2 copies: ORT input/output tensor -> OV tensor (OVEP) -> NPU Tensor (NPU plugin).

### CODE REVIEW

**Comment by sfatimar** [comment]
@jywu-msft @adrianlizarraga

File: onnxruntime/core/providers/openvino/backends/basic_backend.cc [hunk 1]
```diff
@@ -362,23 +349,16 @@ void BasicBackend::StartAsyncInference(Ort::KernelContext& context, OVInferReque
...
-            ov_tensor_data.tensor_ptr = std::make_shared<ov::Tensor>(output.get_element_type(), output.get_shape(),
-                                                                     (void*)tensor.GetTensorRawData());
-          } else {
-            ov_tensor_data.copy_needed = true;
-            ov_tensor_data.tensor_ptr = std::make_shared<ov::Tensor>(output.get_element_type(), output.get_shape());
-          }
-          ort_ov_tensor_map.emplace(ort_tensor_key, ov_tensor_data);
+          ov_tensor_data.ort_ptr = tensor.GetTensorRawData();
+          ov_tensor_data.tensor_ptr = std::make_shared<ov::Tensor>(output.get_element_type(), output.get_shape(),
+                                                                   const_cast<void*>(tensor.GetTensorRawData()));
```
Comment (adrianlizarraga) [inline onnxruntime/core/providers/openvino/backends/basic_backend.cc:360]: non-blocking nit:
Can avoid const_cast by calling `tensor.GetTensorMutableRawData()` on this output tensor. Ort::UnownedValue can return non-const `void*` data: https://github.com/microsoft/onnxruntime/blob/a47254eaef87a4572078d3165f4b9d1c4b4c52b1/include/onnxruntime/core/session/onnxruntime_cxx_api.h#L1484-L1488

**Review by adrianlizarraga — APPROVED** [review]

File: onnxruntime/core/providers/openvino/backends/basic_backend.cc (hunk 1 above)
Comment (jywu-msft) [reply onnxruntime/core/providers/openvino/backends/basic_backend.cc:360]: let's address this in follow up PR

Comment (javier-intel) [reply onnxruntime/core/providers/openvino/backends/basic_backend.cc:360]: Thanks George. I was battling with Git and could not merge the change to address Adrian's comment quickly enough. I'll follow-up with Saurabh to create another PR.

**Pull request merged**